lib/galaxy/tools/__init__.py

```diff
@@ -1310,20 +1269,19 @@ def __parse_trackster_conf(self, tool_source):
         if trackster_conf is not None:
             self.trackster_conf = TracksterConfig.parse(trackster_conf)
 
+    def parse_tests(self):
+        tests_source = self.tool_source
+        if tests_source:
+            try:
+                self.__tests = json.dumps([t.to_dict() for t in parse_tests(self, tests_source)], indent=None)
+            except Exception:
+                self.__tests = None
+                log.exception("Failed to parse tool tests for tool '%s'", self.id)
+
     @property
     def tests(self):
-        self.assert_finalized()
-        if not self.__tests_populated:
-            tests_source = self.__tests_source
-            if tests_source:
-                try:
-                    self.__tests = parse_tests(self, tests_source)
-                except Exception:
-                    self.__tests = None
-                    log.exception("Failed to parse tool tests for tool '%s'", self.id)
-            else:
-                self.__tests = None
-            self.__tests_populated = True
+        if self.__tests:
+            return [ToolTestDescription(d) for d in json.loads(self.__tests)]
         return self.__tests
```

Comment:
```suggestion
        return None
```

seems more explicit.